### ISSUE TITLE
fix --show-lint-violation not working in this example

### ISSUE THREAD

**Issue by spex66** [sqlfluff/sqlfluff#4871]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

## file: "sqlfluff.sql" (with a custom spark-function named `to_metadata_except(..)`)
```sql
SELECT
-- the LT02 issue supposed to fix
`ID Number`,
    map_concat(
        map_from_arrays(
            transform(
                -- causing a
                -- Line 9 Position 78: Found unparsable section: ', *'
                map_values(to_metadata_except(array('key', 'lastUpdatedTime'), *)),
                x -> x
            )
        )
    ) AS metadata
FROM something
```

## running lint reports an "unparsable section"

```bash
✗ sqlfluff lint --dialect=sparksql --rules=LT02 sqlfluff-test.sql
== [sqlfluff-test.sql] FAIL                                                                                                                                                                                             
L:   3 | P:   1 | LT02 | Expected indent of 4 spaces.
                       | [layout.indent]
L:   9 | P:  78 |  PRS | Line 9, Position 78: Found unparsable section: ', *'
WARNING: Parsing errors found and dialect is set to 'sparksql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

## running `fix --show-lint-violation` don't

```bash
✗ sqlfluff fix --show-lint-violations --dialect=sparksql --rules=LT02 sqlfluff-test.sql
==== finding fixable violations ====
== [sqlfluff-test.sql] FAIL                                                                                                                                                                                             
L:   3 | P:   1 | LT02 | Expected indent of 4 spaces.                                                                                                                                                                   
                       | [layout.indent]
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [1 unfixable linting violations found]
==== lint for unfixable violations ====
== [sqlfluff-test.sql] FAIL
L:   3 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
```

### Expected Behaviour

`sqlfluff fix --show-lint-violations` is expected to show the lint errors from `sqlfluff lint`.

Discussed with @tunetheweb on [Slack today](https://sqlfluff.slack.com/archives/C01GPHR8J0G/p1684324961249899?thread_ts=1684316652.220449&cid=C01GPHR8J0G)

### Observed Behaviour

the parameter is expected to show linting results too, which makes it easier to understand why "fix" ends in 
> no fixable linting violations

### How to reproduce

See "What happened" for reproduction steps

### Dialect

sparksql

### Version

```bash
✗ sqlfluff version
2.1.0

➟  python --version
Python 3.11.3
```

### Configuration

no `.sqlfluff` or pyproject.toml section

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @spex66 , thanks for submitting this issue.

with the latest `sqlfluff` the above statement parses properly.

Also pls note that PRS issues (unparsable) are NOT the same as lint violations, so adding `--show-lint-violations` won't show you the parse errors.
